### PR TITLE
Tunnel Termination ACL VPP Plugin

### PR DESCRIPTION
**Pull request by AkeelAli**
Overview:
--------
This PR introduces a custom VPP plugin named `tunterm_acl` into Sonic-VPP. This plugin provides the required dataplane ACL functionality to support Smart Switch HA. In particular, it allows v4-vxlan decap'ed packets to be redirected based on their inner DST IP.

Changes:
--------
1. Adds the `tunterm_acl` plugin source code (including tests and docs) under `platform/vpp/vppbld/plugins/`
2. Updates the `vppbld` Makefile to allow for custom plugins to be copied to & built with VPP
3. Enables the `tunterm_acl` plugin in the `startup.conf` files
4. Adds a change in the `vpp.patch` file to export `vxlan_main`

Plugin:
-----
Details of the `tunterm_acl` plugin implementation can be found in the [README.rst](https://github.com/yue-fred-gao/sonic-platform-vpp/blob/tunterm_acl_plugin/platform/vpp/vppbld/plugins/tunterm_acl/docs/README.rst) file.

### CODE REVIEW

File: platform/vpp/vppbld/Makefile
Comment (yue-fred-gao) [inline platform/vpp/vppbld/Makefile:26]: I think we should move plugins directory into vppbld, which is for building vpp. So everything for vpp build is under the same directory.

Comment (AkeelAli) [reply platform/vpp/vppbld/Makefile]: Sounds good. I moved the plugins under vppbld in https://github.com/sonic-net/sonic-platform-vpp/pull/114/commits/935bb7992331e939ac92348a3ed225e1873aaa34.

File: platform/vpp/plugins/tunterm_acl/tunterm_acl_api.c
Comment (AkeelAli) [reply platform/vpp/plugins/tunterm_acl/tunterm_acl_api.c]: Yes. I updated the comment and the README file to add this as enhancement/TODO item. It's also mentioned in the api file.

File: platform/vpp/plugins/tunterm_acl/tunterm_acl_api.c
Comment (AkeelAli) [reply platform/vpp/plugins/tunterm_acl/tunterm_acl_api.c]: Good catch. Fixed in latest commit.

File: platform/vpp/plugins/tunterm_acl/tunterm_acl_decap.c
Comment (AkeelAli) [reply platform/vpp/plugins/tunterm_acl/tunterm_acl_decap.c]: These loops implement the single/dual/quad loop packet processing pattern used in VPP nodes for performance optimization. See this link for more details: https://fd.io/docs/vpp/v2005/gettingstarted/developers/vnet#effective-graph-dispatch-function-coding. You'll notice my tunterm-acl node doesn't have these loops yet as performance wasn't a concern/priority for us. It's part of the enhancements lists in the README file (the node can be evolved to align with this model if needed).

**Review by yue-fred-gao — APPROVED** [review]
Looks good from SONIC side. Let's wait for Abdel to review from vpp

**Review by abdbaig — CHANGES_REQUESTED** [review]
If not already done, can you use the plugin source code within vpp and run the following to make sure all the code is validated against vpp guidelines:

make checkstyle
make checkstyle-api
make chekcstyle-python

The code may have to committed in local branch for the scripts to work.

File: platform/vpp/vppbld/plugins/tunterm_acl/docs/README.rst
Comment (AkeelAli) [reply platform/vpp/vppbld/plugins/tunterm_acl/docs/README.rst]: Clarified that this is a redirect via a VPP Fib path (as declared in the API using vl_api_fib_path_t)

File: platform/vpp/vppbld/plugins/tunterm_acl/tunterm_acl_api.c
Comment (AkeelAli) [reply platform/vpp/vppbld/plugins/tunterm_acl/tunterm_acl_api.c]: Removed in latest commit (no longer offset, masking directly from start of DIP)

File: platform/vpp/vppbld/plugins/tunterm_acl/tunterm_acl_api.c
Comment (AkeelAli) [reply platform/vpp/vppbld/plugins/tunterm_acl/tunterm_acl_api.c]: Removed in latest commit (no longer skipping vectors, masking directly from start of DIP)

File: platform/vpp/vppbld/plugins/tunterm_acl/tunterm_acl_node.c
Comment (AkeelAli) [reply platform/vpp/vppbld/plugins/tunterm_acl/tunterm_acl_node.c]: Added support for vlan tagged inner packet along with testing.

**Review by abdbaig — APPROVED** [review]

**Pull request merged**
